terraform/azure/main.tf

```diff
@@ -117,7 +117,7 @@ resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
 resource "azurerm_kubernetes_cluster_node_pool" "dask_pool" {
   # If dask_nodes is set, we use that. If it isn't, we use notebook_nodes.
   # This lets us set dask_nodes to an empty array to get no dask nodes
-  for_each = try(var.dask_nodes, var.notebook_nodes)
+  for_each = length(var.dask_nodes) == 0 ? var.notebook_nodes : var.dask_nodes
```

Comment:
The reason I used `try` instead in https://github.com/2i2c-org/infrastructure/pull/834 is that without that, I can't tell JIL to *not* have any dask nodes - nil or null weren't accepted as variables values, and the default of `{}` meant it wasn't possible to figure out a way to not have any nodes :( Maybe if the default for `dask_nodes` is set to something other than `{}` might help? Not sure.

Reply:
Yeah, I'm not sure how to fix this. Because without the suggested change I'm struggling to tell carbon plan _to create_ the nodes. Unless we switch to not automatically creating a dask pool from defined notebook pools

Reply:
Given that terraform is _not_ a procedural language, we may be trying to do something that is too complex for it's capabilities

Reply:
ok, how about we:

1. Remove the one liner setting `dask_nodes = {}` in the JIL tfvars,
2. Merge this as is?

This means that when I `tf apply` JIL later, it'll also create those extra nodepools, but they should be empty and cost nothing. Let's just do that?

Reply:
Seems ok to me for the time being - I've pushed that change